File: php/utility/PreparePath.php
<?php
declare(strict_types=1);

// LongevityCompetition SDK utility: prepare_path

class LongevityCompetitionPreparePath
{
    public static function call(LongevityCompetitionContext $ctx): string
    {
        $point = $ctx->point;
        $parts = [];
        if ($point) {
            $p = \Voxgig\Struct\Struct::getprop($point, 'parts');
            if (is_array($p)) {
                $parts = $p;
            }
        }
        return \Voxgig\Struct\Struct::join($parts, '/', true);
    }
}

File: rb/utility/prepare_path.rb
# LongevityCompetition SDK utility: prepare_path
require_relative 'struct/voxgig_struct'
module LongevityCompetitionUtilities
  PreparePath = ->(ctx) {
    point = ctx.point
    parts = []
    if point
      p = VoxgigStruct.getprop(point, "parts")
      parts = p if p.is_a?(Array)
    end
    VoxgigStruct.join(parts, "/", true)
  }
end
